### PR TITLE
Makefile: fix libpfm4 make test

### LINKED ISSUE #1654: [BUG] need a libpfm4 install script for newbie on linux env
**What happened**:
`make test` failed on linux env local because libpfm4 may not installed on dev environment.

FYI:
https://github.com/google/cadvisor/blob/master/docs/development/build.md#perf-support

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
libpfm4 should be install when run `make test/build` on linux

**Environment**:
- App version:
- Kubernetes version (use `kubectl version`):
- Install details (e.g. helm install args):
- Node environment (for koordlet/runtime-proxy issue):
  - Containerd/Docker version:
  - OS version:
  - Kernal version:
  - Cgroup driver: cgroupfs/systemd
- Others:

### PR DESCRIPTION
**Pull request by bowen-intel**
### Ⅰ. Describe what this PR does

Add libpfm4 build script for Makefile.

### Ⅱ. Does this pull request fix one issue?

fixes #1654 

### Ⅲ. Describe how to verify it

run `make test`

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

**Review by jasonliu747 — APPROVED** [review]

File: hack/libpfm.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+if [ -f /etc/redhat-release ]; then
+    # RedHat or CentOS 
+    sudo yum -y update && \
+        sudo yum -y install cmake wget && \
+        sudo yum -y groupinstall "Development Tools"
+elif [ -f /etc/debian_version ]; then
+    # Debian or Ubuntu 
+    sudo apt-get update && \
+        sudo apt-get -y install bash build-essential cmake wget
+else
```
Comment (zwzhang0107) [inline hack/libpfm.sh:27]: should skip for mac os, and we will add mac env to ci

**Comment by zwzhang0107** [comment]
@saintube

**Review by hormes — APPROVED** [review]

File: hack/libpfm.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    # RedHat or CentOS 
+    sudo yum -y update && \
+        sudo yum -y install cmake wget && \
+        sudo yum -y groupinstall "Development Tools"
+elif [ -f /etc/debian_version ]; then
+    # Debian or Ubuntu 
+    sudo apt-get update && \
+        sudo apt-get -y install bash build-essential cmake wget
+elif [ $(uname) == "Darwin" ]; then
+    echo "skip macOS"
```
Comment (eahydra) [inline hack/libpfm.sh:28]: @bowen-intel hi, here just echo "skip macOS" is not enough, still try to install the libpfm on macOS.

Comment (Rouzip) [reply hack/libpfm.sh:28]: Sorry, I misunderstand @zwzhang0107 comment. If "skip for macOS" is skip libpfm4 install, I will add `exit 0` for macOS.

Comment (bowen-intel) [reply hack/libpfm.sh:28]: Sorry, I misunderstand @zwzhang0107 comment. If "skip for macOS" is for `libpfm4` installation, I will add `exit 0` for macOS.

**Pull request merged**